test/src/test/java/hudson/cli/DisconnectNodeCommandTest.java

```diff
@@ -240,34 +194,40 @@ public void disconnectNodeManyShouldFailIfANodeDoesNotExist() throws Exception {
-        assertThat(slave1.toComputer().getOfflineCause(), instanceOf(OfflineCause.ByCLI.class));
-        assertThat(((OfflineCause.ByCLI) slave1.toComputer().getOfflineCause()).message, equalTo("aCause"));
-        assertThat(slave2.toComputer().isOffline(), equalTo(true));
-        assertThat(slave2.toComputer().getOfflineCause(), instanceOf(OfflineCause.ByCLI.class));
-        assertThat(((OfflineCause.ByCLI) slave2.toComputer().getOfflineCause()).message, equalTo("aCause"));
+        assertOffline(slave1, "aCause");
+        assertOffline(slave2, "aCause");
     }
 
     @Test
     public void disconnectNodeManyShouldSucceedEvenANodeIsSpecifiedTwice() throws Exception {
         DumbSlave slave1 = j.createSlave("aNode1", "", null);
         DumbSlave slave2 = j.createSlave("aNode2", "", null);
-        slave1.toComputer().waitUntilOnline();
-        assertThat(slave1.toComputer().isOnline(), equalTo(true));
-        assertThat(slave1.toComputer().getOfflineCause(), equalTo(null));
-        slave2.toComputer().waitUntilOnline();
-        assertThat(slave2.toComputer().isOnline(), equalTo(true));
-        assertThat(slave2.toComputer().getOfflineCause(), equalTo(null));
+        assertOnline(slave1);
+        assertOnline(slave2);
 
         final CLICommandInvoker.Result result = command
                 .authorizedTo(Computer.DISCONNECT, Jenkins.READ)
                 .invokeWithArgs("aNode1", "aNode2", "aNode1", "-m", "aCause");
         assertThat(result, succeededSilently());
-        assertThat(slave1.toComputer().isOffline(), equalTo(true));
-        assertThat(slave1.toComputer().getOfflineCause(), instanceOf(OfflineCause.ByCLI.class));
-        assertThat(((OfflineCause.ByCLI) slave1.toComputer().getOfflineCause()).message, equalTo("aCause"));
-        assertThat(slave2.toComputer().isOffline(), equalTo(true));
-        assertThat(slave2.toComputer().getOfflineCause(), instanceOf(OfflineCause.ByCLI.class));
-        assertThat(((OfflineCause.ByCLI) slave2.toComputer().getOfflineCause()).message, equalTo("aCause"));
+        assertOffline(slave1, "aCause");
+        assertOffline(slave2, "aCause");
+    }
+
+    private static void assertOnline(DumbSlave slave) throws InterruptedException {
+        var computer = slave.toComputer();
+        computer.waitUntilOnline();
+        assertThat(computer.isOnline(), equalTo(true));
+        assertThat(computer.getOfflineCause(), equalTo(null));
+    }
+
+    private static void assertOffline(DumbSlave slave, String message) {
+        var computer = slave.toComputer();
+        assertThat(computer.isOffline(), equalTo(true));
+        var offlineCause = computer.getOfflineCause();
+        if (offlineCause instanceof OfflineCause.ByCLI cliCause) {
+            assertThat(cliCause.message, equalTo(message));
+        } else {
+            assertThat("seen occasionally in CI", offlineCause, instanceOf(OfflineCause.ChannelTermination.class));
```

Comment:
Do you see this clause when running from your laptop? If so, maybe adjust the message
```suggestion
            assertThat("sometimes overrides ByCLI", offlineCause, instanceOf(OfflineCause.ChannelTermination.class));
```

Reply:
right that makes more sense, it is seen in my laptop too so.